### PR TITLE
[gestures] Properly reset anchor after some gestures

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: #277 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>[gestures] Properly reset anchor after some gestures</changelog>`.

### CODE REVIEW

**Review by tobrun — APPROVED** [review]
nit would have expected a test to be impacted by this change

**Review by pengdev — COMMENTED** [review]
Agree with @tobrun that we need a test to cover it.

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 1]
```diff
@@ -1051,10 +1062,13 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
 
       override fun onAnimationStart(animation: Animator) {}
 
-      override fun onAnimationCancel(animation: Animator) {}
+      override fun onAnimationCancel(animation: Animator) {
+        cameraAnimationsPlugin.anchor = scaleCachedAnchor
+        dispatchCameraIdle()
```
Comment (LukasPaczos) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1067]: Isn't `onAnimationEnd` always following `onAnimationCancel`? Do we need to duplicate the logic here?

**Comment by kiryldz** [comment]
> Agree with @tobrun that we need a test to cover it.

Would also bring a valuable test here but sadly we have `// todo animator testing` in those tests, will create a follow-up task and do that later.

**Review by Chaoba — APPROVED** [review]

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt (hunk 1 above)
Comment (kiryldz) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1067]: https://github.com/mapbox/mapbox-maps-android/pull/280

**Pull request merged**